### ISSUE TITLE
Shared `useQuery` result refetch issue

### ISSUE THREAD

**Issue by hwillson** [apollographql/react-apollo#3131]
Moving this here from the [RA 3 PR](https://github.com/apollographql/react-apollo/pull/2892) thread: https://github.com/apollographql/react-apollo/pull/2892#issuecomment-502784831

cc @patroza

----

When I create a hook with `useQuery`, set `notifyOnNetworkStatusChange: true`, and I use this hook in two different places on the same page, if I call `refetch()`, the `networkStatus` changes for the hook that also provided this `refetch` function, but not for the other one.

Is that by design? Is that easily 'fixable'? At least for me it would be desirable - the same data result is already shared, so it seems counter intuitive that the `networkStatus` updates arent.

----

`const useSomeQuery = () => useQuery(gql`...`, {notifyOnNetworkStatusChange: true})`

Now call `const result = useSomeQuery()` from two different components, call `result.refetch()` inside component1; the `result.networkStatus` will change in component1, but not in 2.
however the `result.data` in both are updated.

If needed I can maybe look into setting up some kind of test app, if you have some suggestions for a good platform to do this, then i'm all ears :)

**Comment by hwillson** [comment]
Hi @patroza - this is by design. Each time you call `useQuery` you're independently subscribing to data updates made in the Apollo Cache. So if `Component1` and `Component2` both call `useQuery`, they'll both start their own subscriptions to watch for data changes. `networkStatus` is used to represent the current status of a query's execution. If you trigger a `refetch` in `Component1` (which leads to its `networkStatus` being updated), this means your `Component1` query is being re-executed. Since the results of that execution will (usually) end up in the cache, `Component2` gets to be completely oblivious to the fact that `Component1` fired a refetch and had its `networkStatus` updated. It hasn't executed its own query, so its `networkStatus` stays as is. `data` on the other hand does get updated for both components (if there were changes in the cache), since both components are watching the cache for data changes. 

So in a nutshell - 2 components using the same query behind the scenes don't care when each other fires that query; they just care about the results.

Thanks!

**Comment by patroza** [comment]
@hwillson thanks a lot! I can imagine this is not the place, but is there a best practice in sharing both?
